src/reason-parser/reason_pprint_ast.ml

```diff
@@ -7106,6 +7157,12 @@ let printer = object(self:'self)
           -> `LastArgIsCallback(callback, List.rev args)
       | _ -> `NormalFunAppl args
     in
+    let formattedFunExpr = match (Reason_ast.processFastPipe funExpr).pexp_desc with
```

Comment:
It's a bit strange to call `processFastPipe` again here. The rewriting/processing of the ast feels a bit intertwined with the pure printing logic.
What if we change the implementation of `processFastPipe` to do the transformation up front?
![image](https://user-images.githubusercontent.com/10634678/42444212-c734aaae-836f-11e8-827a-565ccae80c66.png)